src/datatype/signature.rs

```diff
@@ -130,51 +101,35 @@ impl SignatureType {
-        assert!(SignatureType::RsaSsaPss.sign_msg(b"hello", priv_key).is_ok());
-    }
 
     #[test]
-    fn test_rsa_verify() {
-        let pub_key = "-----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAvK19Xh7Y1zzdjDnXotpx\nLBlDc4oahR98I3YyieKAyPmm3l9R9oZl3HHu9OOA/FVF1/QvwNZbgD9ciLyBGVor\nTNPF/2VZmlQmBF6N3BVkmYF9tF0fu8w2MznCQ9bwHE6JR4oLCsb3H/DSpm/GiQ0n\nWwmeNbWJpVpw5x3j8Tsjc7g7+2PO3e9fqh7gxAoPNj1eGwsiSdG9GVTOTBvsbxQH\n4ZT9lkablCIeMxtIdZtLZ1+LffS+f6qaVf7GCjtmIuo4mFD3BisdyHoLnaSxVSGH\nfRVUSouJPa20nP67PZo6EJoWmEOrqDXtoNASuKfS0BzwftRVl6BR3CCpnyyUbq3y\n7wIDAQAB\n-----END PUBLIC KEY-----";
-        let pub_sig = "BusPdTkDUUG6ISM83snpKt0U8W3cKT8itiTXzJmWLHYBTytwCgW70gTY7z876cj6pELil9nBNC0YIdR4jhy4mu0cEiIIXqN3YWaEzMXAXt+QyjNBzm9POY6y3NF/jvpbZY3wgu5GbOkC6opRPoVARR3K79D9vRIEp6KDLbytqrFuDYd2rSRVfKkbdmt2kCJ17fV6NUWQocseil0kOtKZUI6jz/wt6M3pm5ni1I94Y+mjbwhw6LYEa9qQ++hwg875daaJSkb5jDJ0jkvwb28gsHLjZd1ldX0fuN3alV0RAb91m6klJbxu1ZC1QAfBsK2plkPKH+E4sOhu7NgneHqgFQ==";
+    fn test_rsa_sign_and_verify() {
+        let priv_key = Util::read_file("tests/keys/rsa.der").expect("read priv");
+        let pub_key = pem::parse(Util::read_file("tests/keys/rsa.pub").expect("read pub")).expect("pub_key").contents;
         let msg = b"hello";
-        let key = pem::parse(pub_key).unwrap().contents;
-        let sig = base64::decode(pub_sig).unwrap();
-        assert!(SignatureType::RsaSsaPss.verify_msg(msg, &key, &sig));
-
-        let mut bad_msg = msg.clone();
-        bad_msg[0] = msg[0] ^ 0x01;
-        assert_eq!(false, SignatureType::RsaSsaPss.verify_msg(&bad_msg, pub_key.as_bytes(), &sig));
-
-        let mut bad_sig = sig.clone();
-        bad_sig[0] = sig[0] ^ 0x01;
-        assert_eq!(false, SignatureType::RsaSsaPss.verify_msg(msg, pub_key.as_bytes(), &bad_sig));
+        let sig = SignatureType::RsaSsaPss.sign_msg(msg, &priv_key).expect("sign_msg");
+        let bad_msg = msg.iter().enumerate().map(|(i, b)| if i==0 { b^1 } else { *b }).collect::<Vec<_>>();
```

Comment:
I think this is less clear than the other bit flipping thing I did previously. I would change this back for clarity's sake, but it's not enough for me to block the PR.